frontend_lib/src/helper.js

```diff
@@ -13,25 +16,33 @@ const generateFetchResponse = async fetchResult => {
   }))
 }
 
+const errorFlashMessageTemplateObject = errorMsg => ({
+  type: 'addFlashMsg',
+  data: {
+    msg: <ErrorFlashMessageTemplateHtml errorMsg={errorMsg} />,
+    type: 'danger',
+    delay: 30000
+  }
+})
+
 export const handleFetchResult = async fetchResult => {
-  switch (fetchResult.status) {
-    case 204: return fetchResult
-    case 401:
-      GLOBAL_dispatchEvent({type: 'disconnectedFromApi', date: {}})
-      return generateFetchResponse(fetchResult)
-    case 500:
-      GLOBAL_dispatchEvent({
-        type: 'addFlashMsg',
-        data: {
-          msg: i18n.t('Unexpected error, please inform an administrator'),
-          type: 'danger',
-          delay: undefined
-        }
-      })
-      return generateFetchResponse(fetchResult)
-    default:
-      return generateFetchResponse(fetchResult)
+  const status = fetchResult.status
+
+  if (status === 204) return fetchResult // no result
+  if (status >= 200 && status <= 299) return generateFetchResponse(fetchResult)
+  if (status >= 300 && status <= 399) return generateFetchResponse(fetchResult)
+  if (status === 401) { // unauthorized
+    GLOBAL_dispatchEvent({type: 'disconnectedFromApi', date: {}})
+    return generateFetchResponse(fetchResult)
+  }
+  if (status >= 400 && status <= 499) return generateFetchResponse(fetchResult) // let specific handler handle it with fetchResult.body.code
+  if (status >= 500 && status <= 599) {
+    GLOBAL_dispatchEvent(errorFlashMessageTemplateObject(`Unexpected api error with http status ${status}`)) // no need for translation
```

Comment:
There is no traduction here ?

Reply:
This is the case where we decide to keep it in english maybe ?

Reply:
if yes, you can consider this MR have my approbation.

Reply:
it is